src/utilities/dataplane.ts

```diff
@@ -70,32 +68,36 @@ export function dpTags(dataplane: { networking: DataPlaneNetworking }): LabelVal
     .map(([label, value]) => ({ label, value }))
 }
 
-/*
-getStatus takes Dataplane and DataplaneInsight and returns the status 'Online' or 'Offline'
- */
-export function getStatus(dataplane: { networking: DataPlaneNetworking }, dataplaneInsight: DataPlaneInsight | undefined = { subscriptions: [] }): { status: StatusKeyword, reason: string[] } {
-  const inbounds: TODO = dataplane.networking.inbound ? dataplane.networking.inbound : [{ health: { ready: true } }]
-
-  const errors = inbounds
-    .filter((item: TODO) => item.health && !item.health.ready)
-    .map((item: TODO) => `Inbound on port ${item.port} is not ready (kuma.io/service: ${item.tags['kuma.io/service']})`)
-
-  const subscriptions = dataplaneInsight.subscriptions ? dataplaneInsight.subscriptions : []
-
-  const proxyOnline = subscriptions.some(
-    (item: TODO) => item.connectTime && item.connectTime.length && !item.disconnectTime,
-  )
-
-  let status: StatusKeyword = 'online'
-
-  if (!proxyOnline || errors.length === inbounds.length) {
-    status = 'offline'
-  }
+// getItemStatusFromInsight takes object with subscriptions and returns a
+// status 'online' | 'offline'
+export function getItemStatusFromInsight(insight: { subscriptions: DiscoverySubscription[] } | undefined = { subscriptions: [] }): StatusKeyword {
+  const proxyOnline = (insight.subscriptions ?? []).some((subscription) => subscription.connectTime?.length && !subscription.disconnectTime)
+  return proxyOnline ? 'online' : 'offline'
+}
 
-  if (errors.length > 0) {
-    status = 'partially_degraded'
+// getStatusAndReason takes Dataplane and DataplaneInsight and returns a
+// {status: 'online' | 'offline' | 'partially_degraded', reason: errors[]}
+export function getStatusAndReason(dataplane: { networking: DataPlaneNetworking }, insight: { subscriptions: DiscoverySubscription[] } | undefined = { subscriptions: [] }): { status: StatusKeyword, reason: string[] } {
+  const inbound = dataplane.networking.inbound ?? []
+  const errors = inbound
+    .filter(item => item.health && !item.health.ready)
+    .map(item => `Inbound on port ${item.port} is not ready (kuma.io/service: ${item.tags['kuma.io/service']})`)
+
+  let status: StatusKeyword
+  switch (true) {
```

Comment:
I dataplane.networking.inbound is empty or nil we should have status = "online"

Reply:
I think adding a first:

```ts
case inbound.length == 0:
  status = 'online'
  break
```

is likely the nicest thing you can do here

Reply:
Thanks @lahabana, I added this in, also added a quick bunch of unit-y tests which more or less cover what I'd understood